File: 908-middle-of-the-linked-list/middle-of-the-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        ListNode* n1=head,*n2=head;
        while(n2!=NULL){
            n2=n2->next;
            if(n2==NULL){
                break;
            }
            n2=n2->next;
            n1=n1->next;
        }
        return n1;
    }
};

File: 908-middle-of-the-linked-list/middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        temp1=head
        temp2=head
        while(temp2.next!=None):
            temp1=temp1.next
            temp2=temp2.next
            if(temp2==None):
                break
            temp2=temp2.next
            if(temp2==None):
                break
        return temp1
        